### PR TITLE
Contributions Page Map

### PR DESCRIPTION
**Pull request by danielmmack**
Screen Shot 2020-09-23 at 11 35 13 AM

This PR adds a map button and map to the Contributions page.

**References**
[707](https://github.com/rubyforgood/mutual-aid/issues/707)

### CODE REVIEW

**Comment by danielmmack** [comment]
Just some followup questions:

- What do you guys think of the style of the map? Is there a preference or a default style that you guys had in mind?

- Is there a better place to put the style tag on line 64 in the MapBrowser.vue file?

File: app/javascript/pages/browse/MapBrowser.vue [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  },
+}
+</script>
+
+<style>
+  #map {
+    width: 100%;
+    height: 500px;
+  }
+</style>
```
Comment (exbinary) [inline app/javascript/pages/browse/MapBrowser.vue:69]: This is a great location for these styles. I don't see the `#map` element in the markup here, i assume that brought in by the `mapbox` element?

File: app/javascript/pages/browse/MapBrowser.vue [hunk 2]
```diff
@@ -0,0 +1,69 @@
+<template>
+  <section class="MapBrowser">
+    <mapbox
+        access-token="pk.eyJ1IjoibXV0dWFsLWFpZC1hcHAiLCJhIjoiY2tmZTBvd3UwMDBhbTJ4cDlic2JmMWZoaiJ9.rWscBjdl1SMT5N0yekIJYg"
```
Comment (exbinary) [inline app/javascript/pages/browse/MapBrowser.vue:4]: We'll probably want to pull this in via an environment variable.

That'll require reading that on the server and making it available to the front end, which we don't currently have an example of (well, we have examples of getting page data into vue, but i suspect we'll want a slightly different scheme for globals such as this).

I'm ok with punting on that for now. Would you mind leaving a `FIXME` comment in there so we come back to it (one day!).

File: app/javascript/pages/browse/MapBrowser.vue [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        :map-options="{
+          style: 'mapbox://styles/mapbox/streets-v11',
```
Comment (exbinary) [inline app/javascript/pages/browse/MapBrowser.vue:6]: I don't have a strong preference for one style over another. For reference, i found this list:
https://docs.mapbox.com/api/maps/#mapbox-styles
Of those, i'd be good with `streets`, `light`, `satellite-streets`.
Curious if others have preferences. Also easy to change, so happy to go with your suggestion.

File: app/views/layouts/_head.html.erb [hunk 4]
```diff
@@ -4,6 +4,7 @@
   <%= csp_meta_tag %>
 
   <meta name="viewport" content="width=device-width, initial-scale=1" />
+  <script src='https://api.mapbox.com/mapbox-gl-js/v1.12.0/mapbox-gl.js'></script>
```
Comment (exbinary) [inline app/views/layouts/_head.html.erb:7]: There are obviously tradeoffs in both directions, but there's also the option of packaging this up with webpack:
https://docs.mapbox.com/mapbox-gl-js/api/ and https://docs.mapbox.com/mapbox-gl-js/api/#mapbox-css.

I'm good with using the cdn for now.

File: app/javascript/pages/browse/MapBrowser.vue [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        @map-zoomend="zoomend"
+        @map-click:points="clicked"
+        @geolocate-error="geolocateError"
+        @geolocate-geolocate="geolocate"
+      />
+  </section>
+</template>
+
+<script>
+import Mapbox from 'mapbox-gl-vue'
```
Comment (exbinary) [inline app/javascript/pages/browse/MapBrowser.vue:24]: Awesome, thanks for finding a vue wrapper!

**Review by exbinary — APPROVED** [review]
So sweet! 🔥 . Some comments inline, but looking amazing 🙌🏾.

File: app/javascript/pages/browse/MapBrowser.vue (hunk 1 above)
Comment (maricris-sn) [reply app/javascript/pages/browse/MapBrowser.vue:69]: yes, it's referenced from the mapbox module

File: app/views/layouts/_head.html.erb (hunk 4 above)
Comment (maricris-sn) [reply app/views/layouts/_head.html.erb:7]: we can circle back to this. we first tried adding it via webpack but we encountered problems.

**Review by eabousaif — APPROVED** [review]
LGTM

**Pull request merged**